### ISSUE TITLE
ItemViewModel bind type checking invalid for ObservableList

### ISSUE THREAD

**Issue by PvdBerg1998** [edvin/tornadofx#382]
Adding this issue here as a result of a discussion on Slack.

```Kotlin
class MachineList : App(Main::class) {
    class Main : View("Item View List") {
        val model: model by inject()
        override val root = datagrid(model.machines)
    }
}

class Machine
data class xyz(val machinesProperty: SimpleListProperty<Machine>)

class model : ItemViewModel<xyz>() {
    val machinesProperty = bind { item?.machinesProperty }
    var machines by machinesProperty
}
```

Binding the ListProperty does not work correctly. 

When single stepping through `bind`, it becomes apparent that the facade type detection skips over the ObservableList<*> property. This causes the facade to crash upon usage.

Snippet from the bugged ViewModel code:
```Kotlin
if (facade == null) {
            if (forceObjectProperty) {
                facade = if (value != null) BindingAwareSimpleObjectProperty(this, prop.name, value) else BindingAwareSimpleObjectProperty(this, prop?.name)
            } else {
                facade = when (T::class.javaPrimitiveType ?: T::class) {
                    Int::class.javaPrimitiveType -> if (value != null) BindingAwareSimpleIntegerProperty(this, prop.name, value as Int) else BindingAwareSimpleIntegerProperty(this, prop?.name)
                    Long::class.javaPrimitiveType -> if (value != null) BindingAwareSimpleLongProperty(this, prop.name, value as Long) else BindingAwareSimpleLongProperty(this, prop?.name)
                    Double::class.javaPrimitiveType -> if (value != null) BindingAwareSimpleDoubleProperty(this, prop.name, value as Double) else BindingAwareSimpleDoubleProperty(this, prop?.name)
                    Float::class.javaPrimitiveType -> if (value != null) BindingAwareSimpleFloatProperty(this, prop.name, value as Float) else BindingAwareSimpleFloatProperty(this, prop?.name)
                    Boolean::class.javaPrimitiveType -> if (value != null) BindingAwareSimpleBooleanProperty(this, prop.name, value as Boolean) else BindingAwareSimpleBooleanProperty(this, prop?.name)
                    String::class -> if (value != null) BindingAwareSimpleStringProperty(this, prop.name, value as String) else BindingAwareSimpleStringProperty(this, prop?.name)
                    is ObservableList<*> -> if (value != null) SimpleListProperty(this, prop.name, value as ObservableList<T>) else SimpleListProperty(this, prop?.name)
                    is ObservableSet<*> -> if (value != null) SimpleSetProperty(this, prop.name, value as ObservableSet<T>) else SimpleSetProperty(this, prop?.name)
                    is List<*> -> if (value != null) SimpleListProperty(this, prop.name, (value as List<T>).observable()) else SimpleListProperty(this, prop?.name)
                    is Set<*> -> if (value != null) SimpleSetProperty(this, prop.name, (value as Set<T>).observable()) else SimpleSetProperty(this, prop?.name)
                    else -> if (value != null) BindingAwareSimpleObjectProperty(this, prop.name, value) else BindingAwareSimpleObjectProperty(this, prop?.name)
                }
            }
        }
```

The `is ObservableList<*>` never is true, so this `when` will always default to `BindingAwareSimpleObjectProperty` with a null list.
This causes a NPE upon editing the data, for example:

`model.machines.addAll( ... )`

**Comment by edvin** [comment]
`SimpleListProperty` is not a subclass of `ObservableList` so it shouldn't match? :)

**Comment by PvdBerg1998** [comment]
`bind { item?.machines }` does not compile though

**Comment by edvin** [comment]
No, because `item` is of type `xyz`, which doesn't define a field with that name :)

**Comment by PvdBerg1998** [comment]
Reopening after conversion with @t-boom

**Comment by PvdBerg1998** [comment]
Example ItemViewModel : 
![model](https://user-images.githubusercontent.com/27141525/28716058-63512c9a-739b-11e7-821c-6cfd93355d11.PNG)

Debugging type check:
![not seen in switch](https://user-images.githubusercontent.com/27141525/28716053-5c3c626c-739b-11e7-9d24-3ee2f3a46515.PNG)

The ObservableList has been skipped.

**Comment by HaasJona** [comment]
OK, the `when` argument is a Class, for example a `Class<ObservableList<*>>`. because of that `is Class<*>` would be true, not `is ObservableList<*>`.

**Comment by edvin** [comment]
This is fixed and in the 1.7.9 release :)
